src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClientProxy.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using NLog;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Indexers.Tribler;
+
+namespace NzbDrone.Core.Download.Clients.Tribler
+{
+    public interface ITriblerDownloadClientProxy
+    {
+        ICollection<Download> GetDownloads(TriblerDownloadSettings settings);
+
+        ICollection<File> GetDownloadFiles(TriblerDownloadSettings settings, Download downloadItem);
+
+        GetTriblerSettingsResponse GetConfig(TriblerDownloadSettings settings);
+
+        void RemoveDownload(TriblerDownloadSettings settings, DownloadClientItem item, bool deleteData);
+
+        string AddFromMagnetLink(TriblerDownloadSettings settings, AddDownloadRequest downloadRequest);
+    }
+
+    public class TriblerDownloadClientProxy : ITriblerDownloadClientProxy
+    {
+        protected readonly IHttpClient _httpClient;
+        private readonly Logger _logger;
+
+        public TriblerDownloadClientProxy(IHttpClient httpClient, Logger logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        private HttpRequestBuilder getRequestBuilder(TriblerDownloadSettings settings, string relativePath = null)
+        {
+            var requestBuilder = new HttpRequestBuilder(GetBaseUrl(settings, relativePath))
+                .Accept(HttpAccept.Json);
+
+            requestBuilder.Headers.Add("X-Api-Key", settings.ApiKey);
+
+            requestBuilder.LogResponseContent = true;
+
+            return requestBuilder;
+        }
+
+        public string GetBaseUrl(TriblerDownloadSettings settings, string relativePath = null)
+        {
+            var baseUrl = HttpRequestBuilder.BuildBaseUrl(settings.UseSsl, settings.Host, settings.Port, settings.UrlBase);
+            baseUrl = HttpUri.CombinePath(baseUrl, relativePath);
+
+            return baseUrl;
+        }
+
+        private T ProcessRequest<T>(HttpRequestBuilder requestBuilder)
+            where T : new()
+        {
+            return ProcessRequest<T>(requestBuilder.Build());
+        }
+
+        private T ProcessRequest<T>(HttpRequest requestBuilder)
+            where T : new()
+        {
+            var httpRequest = requestBuilder;
+
+            HttpResponse response;
+
+            _logger.Debug("Url: {0}", httpRequest.Url);
+
+            try
+            {
+                response = _httpClient.Execute(httpRequest);
```

Comment:
Can just define and set and handle the deserialization in the try block

```suggestion
                var response = _httpClient.Execute(httpRequest);
                
                return Json.Deserialize<T>(response.Content);
```